project-metrics/metrics_service/apis/travis.py

```diff
@@ -26,22 +28,36 @@ def __init__(self, status_code: int, err_msg: Text):
 class TravisApi(agithub_base.API):
   """Travis API interface."""
 
-  def __init__(self, *args, **kwargs):
+  # Default/maximum page size from the API.
+  TRAVIS_PAGE_SIZE = 25
 
+  def __init__(self, *args, **kwargs):
     extra_headers = {
         'User-Agent': 'AMPProjectMetrics/1.0.0',
-        'Accept': 'application/vnd.travis-ci.2.1+json',
-        'Content-Type': 'application/json'
+        'Content-Type': 'application/json',
     }
+
     props = agithub_base.ConnectionProperties(
         api_url='api.travis-ci.org',
         secure_http=True,
         extra_headers=extra_headers)
     self.setClient(agithub_base.Client(*args, **kwargs))
     self.setConnectionProperties(props)
 
-    self._token = self._get_travis_token(env.get('GITHUB_API_ACCESS_TOKEN'))
+    try:
+      self.api_token = env.get('TRAVIS_API_ACCESS_TOKEN')
+    except KeyError:
+      logging.warn('No Travis API key found; exchanging GitHub token')
+      self.api_token = self._get_travis_token(
+          env.get('GITHUB_API_ACCESS_TOKEN'))
+
     extra_headers['Authorization'] = 'token %s' % self._token
```

Comment:
Move the whole block that gets the token above where you define `extra_headers`, and add the `Authorization` header as a field in the initializer of `extra_headers`

If you can initialize the whole dict in one go, it makes this a little more readable and makes procedural sense:

`"get info required to init the headers" → "init the headers" → "use headers to init the connection"`

compared with

`"init partial headers" → "get info required to complete the headers" → "complete the headers" → "use headers to init the connection"`

Reply:
Can't do that for the same reason as above; the V2 API query requires that the client be working/set-up, with headers, then a slightly different set of headers are used to make GraphQL queries.

Reply:
Ah I see. Add a comment for this then :)